datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -170,7 +171,16 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
         auth = None
         if config['password']:
             if config['username']:
-                auth = (config['username'], config['password'])
+                if config['auth_type']:
+                    if config['auth_type'] == 'basic':
+                        auth = requests.auth.HTTPBasicAuth(config['username'], config['password'])
+
+                    elif config['auth_type'] == 'digest':
+                        auth = requests.auth.HTTPDigestAuth(config['username'], config['password'])
+
+                else:
+                    auth = (config['username'], config['password'])
```

Comment:
For info: This is the same as basic auth https://3.python-requests.org/user/authentication/#basic-authentication

Reply:
keeping it as is for now as it breaks numerous tests for other checks that are expecting auth to be a tuple and not a basic auth instance

Reply:
See before, this else block can be removed to simplify the logic